### DISCUSSION TITLE
Beam functions for beam or beam elements?

### DISCUSSION THREAD

**Discussion by felixcrazzolara** [idaholab/moose#27549] [Q&A Modules: Solid mechanics] [answered]
Are the utility functions for the Timoshenko beam like [InertialForceBeam](https://mooseframework.inl.gov/source/kernels/InertialForceBeam.html) for the entire beam or a beam element? For example, if I use a 1D mesh with `nx=10`, do I need to provide `Iy` and `Iz` for the entire beam or for just 1/10nth of the beam for `InertialForceBeam`?

**Comment by GiudGiud** [comment]
@cbolisetti

**Reply by cbolisetti** [reply]
They're block restricted. So if you enter a function that varies Ix, Iy in space, it will be used for the elements in the whole block. The block can have any number of elements you want.

**Reply by felixcrazzolara** [reply]
@cbolisetti Thank you for your answer. To be honest though, I din't quite understand what you mean. Suppose I create a mesh using
```
[Mesh]
  [mesh]
    type = GeneratedMeshGenerator
    dim = 1
    nx = 10
    xmin = 0.0 # [m]
    xmax = 1.6 # [m]
  []
[]
```
and I want to model a beam of length 1.6m. Just so that we use the some notation. This beam would have 10 elements, correct? This beam has just one Ix, one Iy and one Iz. Each element (having the same length 0.16m), would have its own set of values Ix,Iy,Iz. What do the values `Iy`,`Iz` that I pass to `InertialForceBeam` refer to?

Somehow I would assume that if I pass just one value (`Iy = 7.33653e-8`), it applies it to all of them? However, there is no error when I pass two values (`'7.33653e-8 6.33653e-8'`), which doesn't make any sense. So, I am quite confused how this all adds up.

Your help would be much appreciated.

**Reply by cbolisetti** [reply]
Looks like I misunderstood your question. If you use one number in this case, it will be applied to all 10 elements. In this case you only have one block, which is the default. I think if you enter multiple values they will be applied to multiple blocks. I am not sure why there's no error thrown in this case..probably an oversight.

**Reply by felixcrazzolara** [reply]
@cbolisetti thank you again for your answer. So to conclude this. If I use a single value for Iy, for example. Does that value refer to the Iy for one element out of the ten (in the above example) or for the entire beam?

**Reply by cbolisetti** [reply, marked as answer]
In your case, that should apply for the entire beam.
